src/client/extension.ts

```diff
@@ -192,13 +205,39 @@ export async function activate(
     }
 
     _telemetry.sendTelemetryEvent("activated");
+}
 
+export function setupAuthFileWatcher() {
+    const watchPath = GetAuthProfileWatchPattern();
+    if (watchPath) {
+        const watcher = vscode.workspace.createFileSystemWatcher(watchPath);
+        watcher.onDidChange(() => handleOrgChange());
+        watcher.onDidCreate(() => handleOrgChange());
+        watcher.onDidDelete(() => handleOrgChange());
+    }
+}
+
+async function handleOrgChange() {
+    const cliContext = new CliAcquisitionContext(_context, _telemetry);
+    const cli = new CliAcquisition(cliContext);
+    const cliPath = await cli.ensureInstalled();
+    const pacContext = new PacWrapperContext(_context, _telemetry);
+    const interop = new PacInterop(pacContext, cliPath);
+    const pacWrapper = new PacWrapper(pacContext, interop);
+    const pacActiveOrg = await pacWrapper?.activeOrg();
+    if (pacActiveOrg && pacActiveOrg.Status === PAC_SUCCESS) {
+        const orgID = pacActiveOrg.Results.OrgId;
+        const artemisResponse = await fetchArtemisResponse(orgID, _telemetry);
+        if (artemisResponse) {
+            const { geoName } = artemisResponse[0];
+            oneDSLoggerWrapper.instantiate(geoName);
+        }
+    }
```

Comment:
Duplicated code from `src/common/copilot/PowerPagesCopilot.ts`. 
- Filewatcher can be limited by the OS, so we should generally not have duplicates for the same file
- PAC calls are generally slow (and occur serially in a queue) so we should not be multiple calls for the same information

Reply:
Abstracted file watcher to a common re-usable function

Reply:
Create this tracking item related to optimizing activeOrg PAC call.
[https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3830140](https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3830140)

Reply:
Changes for activeOrg call optimization have been pushed to this PR itself.